### PR TITLE
MarketAnalysis function + various small changes

### PR DESCRIPTION
**Pull request by GiuseppeMascolo**
Created the Market Analysis function and other sub functions.
Other small chenges at Heuristic_Test, autocorrelation_ACF and histogram_distribution_fit has been made

### CODE REVIEW

File: functions/Heuristic_Test.m [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+%        sd_result: true if the standard deviation of sample 2 is within the interval of confidence
+%        correlation_signal: true if data present correlation with the 1-lag data
 % this function performs simple invariance (i.i.d.) tests on a time series
 % 1. it checks that the variables are identically distributed by looking at the 
 %    histogram of two subsamples
 % 2. it checks that the variables are independent by looking at the 1-lag scatter plot
 % under i.i.d. the location-dispersion ellipsoid should be a circle
 % see "Risk and Asset Allocation"-Springer (2005), by A. Meucci
 
-%% Adding sub-functions
```
Comment (ashpace) [inline functions/Heuristic_Test.m:10]: Why did you delete the subfunctions path? how would it get the sub_functions now?

File: functions/MarketAnalysis.m [hunk 2]
```diff
@@ -0,0 +1,120 @@
+function [iid, Rho, nu, GARCHprop] = MarketAnalysis(comp, plotAutocorr, doHystogramFit, plotFatTails, plotHeuristicTest)
+% Author : Giuseppe Mascolo
```
Comment (ashpace) [inline functions/MarketAnalysis.m]: very nice introduction

File: functions/MarketAnalysis.m [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    end
+    
+    %% Heuristic Test
+    % performs simple invariance (i.i.d.) tests on a time series
+    heuristicResult = Heuristic_Test(compound, name, plotHeuristicTest);
+    
+    %% Distribution Fit
+    %performs tStudent fit or emirical fit and
+    %the valuation of the cdf in our data points, obtaining the grades for the future copula fit.
+    grades = tStudentFitGrades(compound, grades);
```
Comment (ashpace) [inline functions/MarketAnalysis.m]: tStudentFitGrades should not take the previous grades and add up. The job of tStudentFitGrades is to give the grades. The right method is 
grades = [grades ,  tStudentFitGrades(compound)]
Secondly, we need the marginal distributions. The copula without marginal distributions is just the joint feature of the distribution. marginals and copula together is the joint distribution. 
F_X(Qx1(x1),Qx2(x2), ...,QxN(xN)) = F_U(u1,u2,u3,..., uN) (2.31) Meucci
However you are disregarding the marginals and just saving the grades.

**Review by ashpace — CHANGES_REQUESTED** [review]
Generally amazing. However needs backtesting and some additional changes

File: functions/Heuristic_Test.m (hunk 1 above)
Comment (GiuseppeMascolo) [reply functions/Heuristic_Test.m:10]: In order not to have Warnings I moved the path in the main

File: functions/MarketAnalysis.m (hunk 3 above)
Comment (GiuseppeMascolo) [reply functions/MarketAnalysis.m]: I'll fix tStudentFitGrades and the marginals.

**Comment by GiuseppeMascolo** [comment]
Backtasting still to do.
